src/test/java/com/gavinfenton/quizolation/service/TeamServiceTest.java

```diff
@@ -0,0 +1,115 @@
+package com.gavinfenton.quizolation.service;
+
+import com.gavinfenton.quizolation.entity.Team;
+import com.gavinfenton.quizolation.repository.TeamRepository;
+import org.hibernate.ObjectNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class TeamServiceTest {
+
+    @InjectMocks
+    private TeamService teamService;
+
+    @Mock
+    private TeamRepository teamRepository;
+
+    @BeforeEach
+    public void setup() {
+        initMocks(this);
+    }
+
+    @Test
+    public void testCreateTeam() {
+        Team teamSaving = new Team();
+        teamSaving.setName("Team name");
+        Team teamExpected = new Team();
+
+        given(teamRepository.save(teamSaving)).willReturn(teamExpected);
+
+        Team teamActual = teamService.createTeam(teamSaving);
+
+        verify(teamRepository).save(teamSaving);
+        assertEquals(teamExpected, teamActual);
+    }
+
+    @Test
+    public void testUpdateTeam() {
+        Team teamSaving = new Team();
+        teamSaving.setId(1L);
+        teamSaving.setName("Team name");
+        Long idExpected = teamSaving.getId();
```

Comment:
Mixing saving and expected entities could pollute the tests, better to keep them separate so that you can be sure the one returned is from the repo and not the one passed in (i.e. could pass if update() does nothing but return)

Reply:
Still mixing values of saving and expected, I would just remove the setting of ID completely as we don't need it for the test. Just pass in a magic number to update call in then().

Creating two Teams and changing the name of one will make sure that saving and expected don't equals() each other, which is what makes the assertEquals(teamActual, teamExpected); valid as they could only equal if the repo returned teamExpected and not teamSaving (as it has a different name looked at in equals())